### PR TITLE
DSL validator tests

### PR DESCRIPTION
**Pull request by bd82**
@MathieuAA Please review.

If this basic structure is acceptable I can add many more test cases here.

### CODE REVIEW

File: test/spec/grammar/validator_spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          expect(() => parse(`
+            application {
+              config {
+                applicationType foo
+              }
+            }`)).to.not.throw();
+        });
+      });
+
+      context('in-valid input', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: can just write `invalid`

File: test/spec/grammar/validator_spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          expect(() => parse(`
+            application {
+              config {
+                applicationType 666
+              }
+            }`)).to.throw('applicationType property name must match: /^[a-z]+$/');
+        });
+
+        // applicationType is expecting a lower case name, not just any name
+        it('Will report a syntax error when using upper case name', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: you can drop the capital letters

File: test/spec/grammar/validator_spec.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            application {
+              config {
+                applicationType foo
+              }
+            }`)).to.not.throw();
+        });
+      });
+
+      context('in-valid input', () => {
+        it('Will report a syntax error when using a none name', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: you can drop the capital letters

File: test/spec/grammar/validator_spec.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const expect = require('chai').expect;
+const parse = require('../../../lib/dsl/api').parse;
+
+describe('The JDL Validator', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: There's an actual class for this file, so I'd rather use `JDLSyntaxValidatorVisitor` as provided.

File: test/spec/grammar/validator_spec.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        });
+
+        // applicationType is expecting a lower case name, not just any name
+        it('Will report a syntax error when using upper case name', () => {
+          expect(() => parse(`
+            application {
+              config {
+                applicationType UPPER_CASE_NAME
+              }
+            }`)).to.throw('applicationType property name must match: /^[a-z]+$/');
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: perfect!

File: test/spec/grammar/validator_spec.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+const expect = require('chai').expect;
+const parse = require('../../../lib/dsl/api').parse;
+
+describe('The JDL Validator', () => {
+  context('application configuration', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: `when declaring an application`

File: test/spec/grammar/validator_spec.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    context('APPLICATION_TYPE', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: `and using for applicationType`

File: test/spec/grammar/validator_spec.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      context('valid input', () => {
```
Comment (MathieuAA) [inline test/spec/grammar/validator_spec.js]: `a valid value`

**Comment by bd82** [comment]
Hi @MathieuAA 

I tried to apply the naming convention you recommended.
If there is just a tiny fix or two required just edit it directly on github please and we will speed things up :)

**Comment by MathieuAA** [comment]
Why is the yarn.lock file different?

**Comment by bd82** [comment]
> Why is the yarn.lock file different?

I think because I fixed to fixed range for fs-extra using github web editor so the yarn.lock was not synced to the package.json from the previous commit

**Comment by bd82** [comment]
I hope I fixed the eslint errors now.

**Comment by bd82** [comment]
Reminder to squash this to avoid polluting the history 😄

**Comment by bd82** [comment]
Hi @MathieuAA

Do you want to remove yarn.lock from this? is it worth the hassle or should this just be merged?

**Comment by bd82** [comment]
I dont understand how I closed this 😢

**Comment by MathieuAA** [comment]
Just do an empty commit and it should work I think

**Comment by bd82** [comment]
I forced push a commit that reverted everything, I will sort it out 😢

**Comment by MathieuAA** [comment]
Can't you just reopen one? (PR)

**Comment by bd82** [comment]
Fixed now, I started messing with the git reflog and it caused problems...
I just started from a blank slate and that fixed it
https://github.com/jhipster/jhipster-core/pull/238

**Pull request closed without merging**